ts/main.ts

```diff
@@ -0,0 +1,140 @@
+const $weatherAppDiv = document.querySelector(
+  '[data-view="weather-app"]',
+) as HTMLDivElement;
+const $weatherDisplay = document.querySelector(
+  '[data-view="weather-display"]',
+) as HTMLDivElement;
+const $form = document.querySelector('form') as HTMLFormElement;
+const $locationInput = document.querySelector(
+  'input[type="text"]',
```

Comment:
I would like you to keep things consistent. You are calling this `$form` and you query for the first form element. But your second form is called `$weatherForm2` and you query for the _id_ of `weather-form2`

You have an id of `weather-form` on this one so you should make this the same like
`const $weatherForm = document.querySelector('#weather-form') as HTMLFormElement;